tpm2/encoding_test.go

```diff
@@ -477,3 +477,45 @@ func TestSignEncode(t *testing.T) {
 		}
 	})
 }
+
+func TestTPMLDigestEncode(t *testing.T) {
+	hashA := []byte{0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 255}
+	hashB := []byte{0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 255}
+
+	hl := TPMLDigest{Digests: []tpmutil.U16Bytes{
+		hashA, hashB},
+	}
+
+	want := []byte{0, 0, 0, 2, 0, 20, 0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 255,
+		0, 20, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 255}
+
+	have, err := hl.Encode()
+	if err != nil {
+		t.Errorf("Conversion to []byte failed")
+	}
+	if !bytes.Equal(want, have) {
+		t.Errorf("Have: %v \n Want: %v", have, want)
```

Comment:
Nit: The term `got` is more frequently used than `have`. Its also good to describe what specifically the value represents in the error string, so you can tell whats going on just by the test output without looking at the test code. IE:

```go
hl := TPMLDigest{Digests: []tpmutil.U16Bytes{
	hashA, hashB},
}

want := []byte{0, 0, 0, 2, 0, 20, 0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 255,
	0, 20, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 255}
got, err := hl.Encode()
if err != nil {
	t.Errorf("hl.Encode() failed: %v", err)
}
if !bytes.Equal(got, want) {
	t.Errorf("TPMLDigest{%v}.Encode() = %v, want %v", h1, got, want)
}
```